### PR TITLE
make: add target to list available targets

### PR DESCRIPTION
**Pull request by OlegHahm**
There are a plenty of helpful targets in RIOT's build system but I doubt that most people will ever find them. Maybe this helps a little bit.

### CODE REVIEW

**Comment by LudwigOrtmann** [comment]
In OSX, some strange additions appear:
```
$ make help
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Makefile
all
all-cachegrind
all-debug
...
```

**Comment by OlegHahm** [comment]
Can you test if renaming the target to something like `usage` or `list` helps?

**Comment by LudwigOrtmann** [comment]
In FreeBSD, it's off differently (no gcc available, only gcc47 ;-):
```
$ gmake help
gcc: not found
gcc: not found
Makefile
all
all-cachegrind
all-debug
all-gprof
all-valgrind
```

**Comment by LudwigOrtmann** [comment]
```sh
$ make gelp
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Makefile
all
```

File: Makefile.include [hunk 1]
```diff
@@ -302,3 +302,6 @@ else # RIOT_VERSION
 	$(MAKE) RIOTBASE=$(<D) $(filter-out clean, ${MAKECMDGOALS})
 
 endif
+
+help:
+	@$(MAKE) -qp | awk -F':' '/^[a-zA-Z0-9][^$$#\/\t=]*:([^=]|$$)/ {split($$1,A,/ /);for(i in A)print A[i]}'  | sort | uniq
```
Comment (Kijewski) [inline Makefile.include]:
```
@$(MAKE) -qp | sed -ne 's/\(^[a-z][a-z_-]*\):.*/\1/p' | sort | uniq
```

**Comment by LudwigOrtmann** [comment]
That garbage is a local problem, not related to this PR:
```
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
```

File: Makefile.include (hunk 1 above)
Comment (LudwigOrtmann) [inline Makefile.include]: Still works on OS X

**Comment by OlegHahm** [comment]
addressed comment

File: Makefile.include [hunk 2]
```diff
@@ -302,3 +302,6 @@ else # RIOT_VERSION
 	$(MAKE) RIOTBASE=$(<D) $(filter-out clean, ${MAKECMDGOALS})
 
 endif
+
+help:
+	@$(MAKE) -qp | sed -ne 's/\(^[a-z][a-z_-]*\):.*/\1/p' | sort | uniq
```
Comment (gebart) [inline Makefile.include]: is the `.*` always greedy? 
otherwise: should we add a `$` to the end of the match pattern?

Comment (LudwigOrtmann) [inline Makefile.include]: No need for `$`, patterns are always applied to the complete input string.

**Comment by LudwigOrtmann** [comment]
ACK, please squash.

**Comment by OlegHahm** [comment]
Squashed.

**Pull request merged**